docs/source/generics.rst

```diff
@@ -478,6 +438,45 @@ type variables defined with special keyword arguments ``covariant`` or
    my_box = Box(Cat())
    look_into(my_box)  # OK, but mypy would complain here for an invariant type
 
+.. _type-variable-upper-bound:
```

Comment:
I move upper bounded type vars before value restricted type vars because they're more useful and easier to understand